app/src/main/java/org/schabi/newpipe/util/XmlDocument.java

```diff
@@ -0,0 +1,208 @@
+package org.schabi.newpipe.util;
+
+import org.xmlpull.v1.XmlPullParser;
+import org.xmlpull.v1.XmlPullParserException;
+import org.xmlpull.v1.XmlPullParserFactory;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+
+class XmlDocument {

```

Comment:
Why do we have to implement something like this ourself, can't we use libraries for xml parsing?

Reply:
perhaps